lib/context.js

```diff
@@ -15,6 +15,20 @@ function isExpired() {
   return tokenSet.call(this).expired();
 }
 
+async function refresh(opts) {
+  let { config, req } = weakRef(this);
+  const client = await getClient(config);
+  const oldTokenSet = tokenSet.call(this);
+  const newTokenSet = await client.refresh(oldTokenSet, opts);
```

Comment:
It is entirely possible that a refresh() will only yield the access token and no refresh_token. We should look for that specific scenario and keep the old refresh token value.

Reply:
Good point, thanks!